Hesa add yocto license exprs

hesa opened the pull request:
With these commits we can check more packages with licenses declared by Yocto. This is done via `translations` (think aliases).

hesa commented:
Looks like the `reuse` related thingie is back
```
AttributeError: 'TypeVar' object attribute '__doc__' is read-only
Error: Process completed with exit code 1.
```

hesa commented:
Works here.
There's a return value error here when running tests locally. This is not related to this issue/pr so I am leaving that for now.

priv-kweihmann commented:
could you please fix the typo in the last commit and maybe add a reference from where you extracted the data.
I guess that would be good for future references

hesa committed:
Adding Yocto translations/aliases

Adding Yocto's mapping (license name -> SPDX) from:
http://git.yoctoproject.org/cgit/cgit.cgi/poky/plain/meta/conf/licenses.conf

hesa commented:
Added all Yocto translations/alises: http://git.yoctoproject.org/cgit/cgit.cgi/poky/plain/meta/conf/licenses.conf

Reuse problem remains - I say ignore 
Git lint complains, but it is the url, so I say ignore

priv-kweihmann commented:
The reuse error likely goes away if you would rebase onto latest main
